### PR TITLE
Allow finite replay

### PR DESCRIPTION
**Pull request by jcare44**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ Replay messages from (RabbitMQ) AMQP dead-letter queue
 ```
 # add this to your [bash|zsh]rc and source it
 function amqp-replay(){
-  docker run -it -e AMQP_URI=$AMQP_URI -e AMQP_QUEUE_NAME=$AMQP_QUEUE_NAME -e AMQP_QUEUE_NOACK=$AMQP_QUEUE_NOACK -e AMQP_EXCHANGE_NAME=$AMQP_EXCHANGE_NAME --rm fgribreau/amqp-replay:v0.1.0
```
Comment (jcare44) [inline README.md:20]: It should be better to use `latest` right ? What do you think ?

File: README.md [hunk 2]
```diff
@@ -34,6 +34,8 @@ npm i amqp-replay -g
 AMQP_URI="amqp://user:password@rabbitmq.ndd.com:5672/%2F" AMQP_QUEUE_NAME="social.test" AMQP_EXCHANGE_NAME="fgtest" amqp-replay
 ```
 
+You can add `INFINITE=true` for an infinite replay of the queue.
```
Comment (FGRibreau) [inline README.md:37]: Should we consider INFINITE to be the "good default" or not?

File: amqp-replay.js [hunk 3]
```diff
@@ -43,13 +45,39 @@ require('amqplib').connect(config.amqp.uri).then(function(conn) {
...
         }
-      }, {
-        noAck: config.amqp.noAck
-      });
+      }
+
+      function getAllMessagesInQueue(res = []) {
+        return queueCh.get(config.amqp.queue.name, {noAck: config.amqp.noAck})
+          .then(newVal => {
+            if (newVal === false) {
```
Comment (FGRibreau) [inline amqp-replay.js]:
```
const shouldStop = newVal === false; // amqp states that...
if(shouldShop){
}
```

File: amqp-replay.js [hunk 4]
```diff
@@ -43,13 +45,39 @@ require('amqplib').connect(config.amqp.uri).then(function(conn) {
...
+          });
+      }
+
+      if (config.infinite) {
+        //replay all messages in queue, infinitely
+        return queueCh.consume(config.amqp.queue.name, handleMessage, {noAck: config.amqp.noAck})
+          .then(_consumeOk => logger.debug('Waiting for messages. To exit press CTRL+C'))
+      } else {
+        //Get all messages currently in queue, and exit once they are replay
+        return getAllMessagesInQueue().then(messages => {
```
Comment (FGRibreau) [inline amqp-replay.js]: quid when the queue is about 1 000 000 messages ? We should have a default batch value (customizable through env. variables)

File: amqp-replay.js [hunk 5]
```diff
@@ -43,13 +45,39 @@ require('amqplib').connect(config.amqp.uri).then(function(conn) {
...
+            return getAllMessagesInQueue(res);
+          });
+      }
+
+      if (config.infinite) {
+        //replay all messages in queue, infinitely
+        return queueCh.consume(config.amqp.queue.name, handleMessage, {noAck: config.amqp.noAck})
+          .then(_consumeOk => logger.debug('Waiting for messages. To exit press CTRL+C'))
+      } else {
+        //Get all messages currently in queue, and exit once they are replay
```
Comment (FGRibreau) [inline amqp-replay.js]: `// get`

File: amqp-replay.js [hunk 6]
```diff
@@ -43,13 +45,39 @@ require('amqplib').connect(config.amqp.uri).then(function(conn) {
...
+              return res;
+            }
+
+            res.push(newVal);
+            return getAllMessagesInQueue(res);
+          });
+      }
+
+      if (config.infinite) {
+        //replay all messages in queue, infinitely
```
Comment (FGRibreau) [inline amqp-replay.js]: `// replay`

File: README.md (hunk 2 above)
Comment (jcare44) [reply README.md:37]: I think that, most of the time, you only want to replay dead lettered messages. So false by default is the most logical IMHO :)

Comment (FGRibreau) [reply README.md:37]: good :+1:

File: README.md (hunk 1 above)
Comment (FGRibreau) [reply README.md:20]: `Always prefer explicit over implicit`

File: amqp-replay.js [hunk 7]
```diff
@@ -43,13 +46,40 @@ require('amqplib').connect(config.amqp.uri).then(function(conn) {
...
         }
-      }, {
-        noAck: config.amqp.noAck
-      });
+      }
+
+      function getAllMessagesInQueue(res = []) {
+        return queueCh.get(config.amqp.queue.name, {noAck: config.amqp.noAck})
+          .then(newVal => {
+            const shouldStop = newVal === false || res.length >= config.bufferLimit;
```
Comment (FGRibreau) [inline amqp-replay.js]: document rational (a-k-a add comments, explaining this)

**Pull request merged**